git-multimail/README

```diff
@@ -388,6 +394,18 @@ multimailhook.replyToRefchange
     - The value "none", in which case the Reply-To: field will be
       omitted.
 
+multimailhook.refFilterInclusionRegex
+multimailhook.refFilterExclusionRegex
+
+    Regular expressions that can be used to limit refs for which email
+    updates will be sent.  It is an error to specify both an inclusion
+    and an exclusion regex.  If an inclusion regex is specified, emails
+    will only be sent for refs which match this regex.  If an exclusion
+    regex is specified, emails will be sent for all refs except those
+    that match this regex (or that match a predefined regex specific
+    to the environment, such as "^refs/notes" for most environments and
+    "^refs/notes|^refs/changes" for the gerrit environment).
```

Comment:
newren <notifications@github.com> writes:

> Actually, neither of these variables is a set/list/whatever of
> regexes.

Currently, yes. My point is that it would be easier to use if it became
a list of regexp. Just like `grep` can accept multiple `-e` options so
you can write

    grep -e foo -e bar

instead of

    grep -e 'foo\|bar'

I find

    [multimailhook]
        refFilterExclusionRegex = ^refs/notes
        refFilterExclusionRegex = ^refs/changes

easier to read than

    [multimailhook]
        refFilterExclusionRegex = ^refs/notes\|^refs/changes

plus I don't have to remember which kind of regurar expression, what
kind of quoting I should use and all in the first case (| Vs \| Vs \\|
depending on the context).

And I could give more complex examples like

    [multimailhook]
        refFilterExclusionRegex = ^refs/heads/\(foo\|bar\)/boz\|^refs/changes

which get really hard to read when hand-assembled into a single regular
expression, even though the expressivity is the same.

Now, the good news is that both "space separated list of regexp" and
"multi-valued config variable" can be added later as an extension to
your version. The multi-valued config variable system would break the
compatibility only for users relying on the "last one wins" semantics.

> Clearly, my wording didn't have the intended effect. However, I like
> the fact that it's just a regex. If we make it a space separated list
> or multiple config values, then it loses that quality.

Sorry, but I see no real argument here. If you like writting single
regexp, you'll still be able to do so.

-- 
Matthieu Moy
http://www-verimag.imag.fr/~moy/